docs/developer/release-process.md

```diff
@@ -2,67 +2,154 @@
 
 ```bash
 helm dependency list ./chart/kubeapps
 ```
 
-to see if the latest versions are included, and
+- If they are not, run this other command to update the `requirements.lock` file:
 
 ```bash
 helm dependency update ./chart/kubeapps
 ```
 
-to update the requirements.lock file.
+> As part of this release process, the chart dependencies _must_ be updated to the latest versions. In case of a major version, the change _should_ be tracked in a separate PR.
+
+### 0.4 - Upgrading the code dependencies
+
+Currently, we have three types of dependencies: the [dashboard dependencies](../../dashboard/package.json), the [golang dependencies](../../go.mod), and the [rust dependencies](../../cmd/pinniped-proxy/Cargo.toml). They _must_ be upgraded to the latest minor/patch version to get the latest bug and security fixes.
 
-Next, run
+- Upgrade the [dashboard dependencies](../../dashboard/package.json) by running:
 
 ```bash
+cd dashboard
 yarn upgrade
 ```
 
-in the dashboard directory will update the frontend packages to the latest compatible versions.
+- Check the outdated [golang dependencies](../../go.mod) by running:
 
-Finally, look at the [pull requests](https://github.com/kubeapps/kubeapps/pulls) and ensure there is no PR open by Snyk fixing a security issue.  If so, discuss it with a peer and come to a decision on it, trying not to release with a high/medium severity issue.
+```bash
+go mod tidy
+go list -u -f '{{if (and (not (or .Main .Indirect)) .Update)}}{{.Path}}: {{.Version}} -> {{.Update.Version}}{{end}}' -m all 2> /dev/null
+```
 
-## 1 - Create a new git tag
+Then, try to manually update those versions that can be safely upgraded. A useful tool for doing so is [go-mod-upgrade](https://github.com/oligot/go-mod-upgrade).
 
-The first step is to tag the repository master branch tip and push it upstream. It is important to note that the tag name will be used as the release name.
+- Upgrade the [rust dependencies](../../cmd/pinniped-proxy/Cargo.toml) by running:
 
 ```bash
-export VERSION_NAME="v1.0.0-beta.1"
+cd cmd/pinniped-proxy/
+cargo update
+```
+
+- Finally, look at the [pull requests](https://github.com/kubeapps/kubeapps/pulls) and ensure there is no PR open by Snyk fixing a security issue. If so, discuss it with a peer and come to a decision on it, trying not to release with a high/medium severity issue.
+
+> As part of this release process, the dashboard deps _must_ be updated, the golang deps _should_ be updated, the rust deps _should_ be updated and the security check _must_ be performed.
+
+## 1 - Send a PR to the bitnami/chart repository
+
+Since the chart that we host in the Kubeapps repository is only intended for development purposes, we need to synchronize it with the official one in the [bitnami/charts repository](https://github.com/bitnami/charts/tree/master/bitnami/kubeapps). To this end, we need to send a PR with the changes to their repository and wait until it gets accepted.
```

Comment:
Given that they're different repos with different directory structures, how did you create the PR (I guess if we branch the bitnami charts and copy our files over we could miss some removals etc., so did you delete all files then copy and compare?)

Reply:
Yep, even if there are proper ways to sync directories, I just ended up deleting and copying the files, TBH. For the automated PR, we should come up with better approaches here: git with multiple remotes, rsyncing dirs, etc